### PR TITLE
[sonic_installer] consider swap checking memory

### PR DESCRIPTION
**Pull request by Xichen96**
Signed-off-by: Xichen Lin <lukelin0907@gmail.com>

#### What I did
Consider preexisting swap when trying to enable swap before installing new image

#### How to verify it
Run script

### CODE REVIEW

File: sonic_installer/main.py [hunk 1]
```diff
@@ -460,8 +460,10 @@ def __enter__(self):
             meminfo = self.read_from_meminfo()
             mem_total_in_bytes = meminfo["MemTotal"] * SWAPAllocator.KiB_TO_BYTES_FACTOR
             mem_avail_in_bytes = meminfo["MemAvailable"] * SWAPAllocator.KiB_TO_BYTES_FACTOR
-            if (mem_total_in_bytes < self.total_mem_threshold * SWAPAllocator.MiB_TO_BYTES_FACTOR
-                    or mem_avail_in_bytes < self.available_mem_threshold * SWAPAllocator.MiB_TO_BYTES_FACTOR):
+            swap_total_in_bytes = meminfo["SwapTotal"] * SWAPAllocator.KiB_TO_BYTES_FACTOR
```
Comment (Blueve) [inline sonic_installer/main.py:463]: Do we always have this value? If no swap exists, what value will be here then?

**Review by Blueve — CHANGES_REQUESTED** [review]
Please update unittest

**Comment by Blueve** [comment]
@lolyu  could you please take a look as well?

File: sonic_installer/main.py (hunk 1 above)
Comment (Xichen96) [reply sonic_installer/main.py:463]: The value is retrieved from /proc/meminfo, we will always have it. If swap was not on, they would be 0.

**Review by Blueve — APPROVED** [review]

File: tests/swap_allocator_test.py [hunk 2]
```diff
@@ -112,7 +116,9 @@ def test_swap_allocator_context_enter_allocate_true_insufficient_total_memory(se
             mock_disk_free.return_value = 10 * 1024 * 1024 * 1024
             mock_meminfo.return_value = {
                 "MemTotal": 2000000,
-                "MemAvailable": 1900000,
+                "MemAvailable": 1800000,
```
Comment (Blueve) [inline tests/swap_allocator_test.py:119]: Seems like this change is unnecessary

**Review by Blueve — CHANGES_REQUESTED** [review]
Please revert unnecessary changes

**Review by lolyu — APPROVED** [review]
LGTM

**Comment by Blueve** [comment]
@Xichen96  please fix below test failure:
```
tests/swap_allocator_test.py:202: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_mock_self = <MagicMock name='setup_swapmem' id='140239347861152'>

    def assert_called_once(_mock_self):
        """assert that the mock was called only once.
        """
        self = _mock_self
        if not self.call_count == 1:
            msg = ("Expected '%s' to have been called once. Called %s times." %
                   (self._mock_name or 'mock', self.call_count))
>           raise AssertionError(msg)
E           AssertionError: Expected 'setup_swapmem' to have been called once. Called 0 times.
```

**Review by Blueve — APPROVED** [review]

**Comment by qiluo-msft** [comment]
@Xichen96 In future, please use squash merge, not normal merge.

**Comment by qiluo-msft** [comment]
The merged 2 commits could not be directly cherry-picked to 202012. Please submit another PR.

**Comment by liuh-80** [comment]
This PR already cherry-pick to 202012 with this PR: https://github.com/sonic-net/sonic-utilities/pull/2360

**Pull request merged**